### DISCUSSION TITLE
[XP Tracker] Automatic Canvas Mode

### DISCUSSION THREAD

**Discussion by Jademalo** [runelite/runelite#15595] [Ideas]
At present, XP Tracker has an incredibly useful piece of functionality where you can right click on the info box for any skill and add it to the game canvas. This is great for keeping the game big and playing with the sidebar collapsed, however is a little inconvenient since you need to add the skill you're training to the canvas manually every time.

What I propose is fairly simple - An option to add Total Gained XP as a canvas option with similar functionality to the native XP drops box.

![image](https://user-images.githubusercontent.com/386846/187710494-986d3c35-8085-49f5-9a1a-024e8e1fdbaf.png)
When selecting add to canvas here, it adds a widget that shows total XP in a similar manner to the native XP Drop tracker. When an XP drop is received, it should automatically switch to show the tracker for that skill, similar to how the native tracker works when Counter and Progress bar are set to "Most Recent".

This would allow you to disable the native tracker entirely and replace it with the much more advanced RuneLite tracker, and get rid of the need to keep adding and removing the tracker for whichever skill you're training.
There could also potentially be an option to disable the automatic switching behaviour for people who want to simply be able to add total XP as a canvas widget.

**Comment by ASong5** [comment]
Bumping this old idea as I still think this would be a great feature

**Comment by Darkforge317** [comment]
Bumping this as well. Was looking for a setting that causes my canvas state to persist, but having the skill get auto-canvased as I earn XP would work just as well!
